### ISSUE TITLE
[New Tracker] Trezzor and CzTorrent

### ISSUE THREAD

**Issue by gregorij89** [Jackett/Jackett#1151]
Hi,
I am trying to code indexer for Trezzor tracker (tracker.czech-server.com) and CzTracker (tracker.cztorrent.net). They are private, respecitve semi-private, Czech trackers. The first one requires standard login form, the second one uses recaptcha 2.

In case of Trezzor, I can add indexer and perform searching. But after some time, the communication is stopped and I have to open indexer settings and click Okay. Then it starts searching again for a while and so on and so on.

In case of CzTracker, it is nearly the same. Leaving aside I can add it only with cookie method (direct login is not working for me yet), it stops working even in shorter period of time than Trezzor. To fix it I have to recreate cookie again.

Why both of them get signed out in some period of time? In case of Trezzor, I have ApplyConfiguration func similar to TransmitTheNet indexer, and in case of CzTracker it is similar to TorrentDay indexer.

Any help will be appreciated.

**Comment by gregorij89** [comment]
Ok, it looks like it was my mistake. My testing server is behind multiwan firewall/router and there was a rule for load balancing network connection for that server. So probably when Jackett initiated the connection to the tracker and firewall sent it through the second interface than login was performed, it was refused and signed off. I've disabled this rule and for now, it seems to work well. I will do some more testing on this issue (and for both indexers in general - they both are connected to Sonarr and Radarr app and for now, it looks good).

Next, I must figure out how to get working reCAPTCHA 2 login. There is some issue with redirecting, because the tracker shows login page, user types credentials and verify he's not a robot, then login process continues to page with the text "You've been successfuly loged in. Please continue HERE" and after a second it is redirected to final internal site. For now I've ended up in a loop, where Jackett tries to sign in and the first redirect returns me to the login page without signing in.

So for now, the Trezzor indexer is working and the CzTracker indexer also, but only with inserted cookie. I'll keep you informed about progress. Cheers

**Comment by kaso17** [comment]
Would have suggested to use cardigann definitions for new trackers but seems like it's already to late for that.

Regarding your redirect issue. Just check for the "You've been successfuly loged in" message in the ConfigureIfOK call (most trackers check for the logout link which probable isn't shown yet).

**Comment by gregorij89** [comment]
Unfortunatelly it is not that simple. I've already checked for that phrase. The problem lies somewhere in the cookie creation.

Cookie created and sended by Jackett (even after calling RequestLoginAndFollowRedirect) is something like
__cfduid=d541dc9362c1dc06cbc1478bde09692061489350184; TestCookie=ok; request_uri=%2Flogin-page%3Furl%3D%252Findex.html

But in browser before login it looks like:
__cfduid=d50febf52e0ade31649f06b5a392c9f631489169436; TestCookie=ok; request_uri=%2Findex.html; level=2; __utma=58721768.1204354109.1489176713.1489271614.1489347413.10; __utmc=58721768; __utmz=58721768.1489176713.1.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none)

And after hitting Log in button:
__cfduid=d50febf52e0ade31649f06b5a392c9f631489169436; TestCookie=ok; level=2; request_uri=%2Flogin-page%3Furl%3D%252Findex.html; pass={SOMEHASH}; uid={SOMEINT}; __utmt=1; __utma=58721768.1204354109.1489176713.1489347413.1489349746.11; __utmb=58721768.2.10.1489349746; __utmc=58721768; __utmz=58721768.1489176713.1.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none)

So somewhere in the Jackett login process, it fails to add new cookie values. I am not so skilled in web page coding and I am slowly getting familiar with Jackett source code, so I am making only small progress during the time. 

PS: The loging out issue was definetly caused by multi wan load balancing. Both indexers stayed logged in during whole day.

PPS: Maybe it is not to late to convert CzTorrent indexer to cardigan definition, if it will solve the login issue. I'll take a look over it in some free time.

Cheers

**Comment by kaso17** [comment]
does RequestLoginAndFollowRedirect return the correct cookies?

**Comment by kaso17** [comment]
If you want to port it to a cardigann definition this login block should work for cztorrents:
```yaml
  login:
    path: /login-page
    method: form
    form: form[action^="/login"]
    inputs:
      username: "{{ .Config.username }}"
      password: "{{ .Config.password }}"
      persistent_login: "1"
    error:
      - selector: div.error
    test:
      path: /torrents
```

**Comment by halali** [comment]
any news ?

**Comment by gregorij89** [comment]
Unfortunatelly not. To busy at work, so no time for developing. Both extensions are working with Sonarr and Radarr, but ...

Both have issue with loging out when user login to tracker page in browser.

And of course with CzTorrent, there is still present the issue with signing in. That is possible only with cookie, not with name, password and reCAPTCHA2. And another one I have discovered is when user is not member, but validating on the tracker, jackett returns forbidden torrents as any other normal torrents, but Sonarr or Radarr is unable to download them - this has consequences to RSS syncs and availability of new releases. I have to apply some sort of filtering before I publicly release the CzTorrent indexer. The goal is to retrun only torrents user can download. To be honest, making the indexer for CzTorrent is harder then I´ve thought.

Cheers

**Comment by gregorij89** [comment]
Hi, still no luck with CzTorrent signing in. But both indexers are doing well, I´ve modified CzTorent to return only valid torrents (it goes through pages before it collects 50 torrents minimal or finds end of the list). But from my point of view, both can be useful for others, signing with a cookie is not so hard and someone who really needs this extension could go through this process. Or maybe when one of you smart guys looks over the code then you maybe find why it is not working.

But I have another problem now. I've never participated in the project on GitHub, so I've my local working build and I see changes in the Visual Studio's Solution Explorer, but I don't know how to upload to code to the project? I know it is a really stupid question, but could someone point me in the right direction?

George

**Comment by JigSawFr** [comment]
@gregorij89 you need to learn how to clone a git repo (in vs) and use git to submit your changes.
Take a look at the great microsoft docs here: https://www.visualstudio.com/en-us/docs/git/gitquickstart :)

**Comment by gregorij89** [comment]
Thank you for the link @JigSawFr. I had a local clone, so I've created a local branch. But any attempt to push or to create new pull request ends with fatal: unable to access 'https://github.com/Jackett/Jackett.git/': The requested URL returned error: 403

**Comment by halali** [comment]
Any news ?

**Comment by JigSawFr** [comment]
@halali you are not authorize to push directly to this repository. To do your need to "fork" si repo and clone your fork to work on. After that, you can open a "pull request" here to submit your changes :)

**Comment by halali** [comment]
@JigSawFr i just ask @gregorij89  for news about CZTorrent tracker regarding to this

> Hi @halali, not until I get approval from tracker management. They are considering it right now and as soon as I get their conclusion, I will eventually create a pull request. I am a member of this tracker and I don´t want to get banned. It is also in your interest to wait a little longer because if they will approve my piece of code, it is unlikely that they will block it in the future. As you probably know, the tracker has issues with overloading and they are concerned about that. As soon as I will have some new info I´ll post it here (or probably in #1151 thread).

>

**Comment by JigSawFr** [comment]
Mea culpa :)

**Comment by halali** [comment]
Hi,

Im trying to make cardigan script for CzTorrent and im stucked, so maybe you can help me.

There is few types of release
`The 100 S03E02 -​ Wanheda, Part 2 (SK)[TvRip]
V utajení /​ Unlocked (2017)[1080pLQ]
Kráska a zvíře /​ Beauty and the Beast (2017)(CZ/​SK/​EN)[HEVC][1080pHD]`
1st part befor / is in czech lang so i tryed filter it out by code
`      title:
        selector: td.detaily a
        filters:
          - name: split
            args: [ "/", 1 ]`
but there is problem when there is more than one /

also when in title is no "/" it wont show in search result

**Comment by kaso17** [comment]
In that case you probably will need a regex filter.
Try
```
      - name: re_replace
        args: [".*? / ", ""]
```
That should remove any test up to the first " / " string.

**Comment by halali** [comment]
it works great

**Comment by garfield69** [comment]
closing. Trezzor and CzTorrent have been added.

**Comment by garfield69** [comment]
https://tracker.cztorrent.net/ and https://cztorrent.net/ are redirecting to https://ems.fcnantes.com/ which presents
```html
403 Forbidden
nginx/1.18.0
```
either using my native IP or via a VPN IP

is the site down, broken, moved to a new domain, or just me?

**Comment by halali** [comment]
I think Czech Police closed this server.

Dňa ut 8. 9. 2020, 6:04 garfield69 <notifications@github.com> napísal(a):

**Comment by ilike2burnthing** [comment]
https://faq.cztorrent.net/ is still up

One of the admins is saying it's temporary and not related to the recent raids - https://twitter.com/Skippy_CZT

There's suggestions elsewhere that they were hacked.

**Comment by halali** [comment]
Hope so

**Comment by garfield69** [comment]
Jackett 0.16.1709
